pkg/api/v1beta3/types.go

```diff
@@ -206,7 +206,10 @@ type HostPathVolumeSource struct {
 	Path string `json:"path" description:"path of the directory on the host"`
 }
 
-type EmptyDirVolumeSource struct{}
+type EmptyDirVolumeSource struct {
+	// Optional: make this empty directory be backed by memory.
+	Tmpfs bool `json:"tmpfs" description:"use tmpfs instead of disk"`
```

Comment:
OK.  Let me think about how to position this so that "tmpfs" is a valid value but it can expand into more QoS-ness later.

I don't want to call the default "Disk" because it might not be.  How about

```
type EmptyDirVolumeSource {
    Medium StorageType
}


type StorageType string

const (
    StorageTypeDefault StorageType = ""  // Or should we make it "Default" and set it in every versioned defaults.go?
    StorageTypeTmpfs StorageType = "Tmpfs"  // Or should we say "Memory", which is a bit more abstract?
)
```